File: api-module/src/main/java/com/shiftmanager/api/service/CalendarService.java
package com.shiftmanager.api.service;

import com.shiftmanager.api.model.Calendar;
import com.shiftmanager.api.model.CalendarEntry;

import java.time.LocalDate;
import java.util.List;

/**
 * Service interface for calendar operations
 */
public interface CalendarService {
    
    /**
     * Get calendar by ID
     * @param calendarId Calendar ID
     * @return Calendar
     */
    Calendar getCalendarById(Long calendarId);
    
    /**
     * Get calendar by employee ID
     * @param employeeId Employee ID
     * @return Calendar
     */
    Calendar getCalendarByEmployeeId(Long employeeId);
    
    /**
     * Get or create calendar for employee
     * @param employeeId Employee ID
     * @return Calendar
     */
    Calendar getOrCreateCalendarForEmployee(Long employeeId);
    
    /**
     * Get entries for a calendar
     * @param calendarId Calendar ID
     * @return List of calendar entries
     */
    List<CalendarEntry> getCalendarEntries(Long calendarId);
    
    /**
     * Get entries for a calendar and date
     * @param calendarId Calendar ID
     * @param date Date
     * @return List of calendar entries
     */
    List<CalendarEntry> getCalendarEntriesForDate(Long calendarId, LocalDate date);
    
    /**
     * Get entries for a calendar and date range
     * @param calendarId Calendar ID
     * @param startDate Start date
     * @param endDate End date
     * @return List of calendar entries
     */
    List<CalendarEntry> getCalendarEntriesForDateRange(Long calendarId, LocalDate startDate, LocalDate endDate);
    
    /**
     * Get entries by entry type
     * @param calendarId Calendar ID
     * @param entryType Entry type
     * @return List of calendar entries
     */
    List<CalendarEntry> getCalendarEntriesByType(Long calendarId, String entryType);
    
    /**
     * Add entry to calendar
     * @param calendarId Calendar ID
     * @param entry Calendar entry
     * @return Created calendar entry
     */
    CalendarEntry addCalendarEntry(Long calendarId, CalendarEntry entry);
    
    /**
     * Update calendar entry
     * @param entryId Entry ID
     * @param entry Calendar entry
     * @return Updated calendar entry
     */
    CalendarEntry updateCalendarEntry(Long entryId, CalendarEntry entry);
    
    /**
     * Delete calendar entry
     * @param entryId Entry ID
     * @return true if deleted
     */
    boolean deleteCalendarEntry(Long entryId);
    
    /**
     * Sync employee shifts to calendar
     * @param employeeId Employee ID
     * @return Number of entries created
     */
    int syncEmployeeShiftsToCalendar(Long employeeId);
    
    /**
     * Sync employee vacation requests to calendar
     * @param employeeId Employee ID
     * @return Number of entries created
     */
    int syncEmployeeVacationRequestsToCalendar(Long employeeId);
    
    /**
     * Get calendars for a manager's team
     * @param managerId Manager ID
     * @return List of calendars
     */
    List<Calendar> getTeamCalendars(Long managerId);
}

File: web-client/src/app/services/calendar.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CalendarEntry } from '../models/calendar-entry.model';

@Injectable({
  providedIn: 'root'
})
export class CalendarService {
  private baseUrl = '/api/calendar';

  constructor(private http: HttpClient) { }

  getPersonalCalendar(startDate?: string, endDate?: string): Observable<CalendarEntry[]> {
    let params = new HttpParams();
    if (startDate) params = params.append('startDate', startDate);
    if (endDate) params = params.append('endDate', endDate);
    
    return this.http.get<CalendarEntry[]>(`${this.baseUrl}/personal`, { params });
  }

  getTeamCalendar(startDate?: string, endDate?: string): Observable<CalendarEntry[]> {
    let params = new HttpParams();
    if (startDate) params = params.append('startDate', startDate);
    if (endDate) params = params.append('endDate', endDate);
    
    return this.http.get<CalendarEntry[]>(`${this.baseUrl}/team`, { params });
  }

  getEmployeeCalendar(employeeId: number, startDate?: string, endDate?: string): Observable<CalendarEntry[]> {
    let params = new HttpParams();
    if (startDate) params = params.append('startDate', startDate);
    if (endDate) params = params.append('endDate', endDate);
    
    return this.http.get<CalendarEntry[]>(`${this.baseUrl}/employee/${employeeId}`, { params });
  }

  createCalendarEntry(entry: CalendarEntry): Observable<CalendarEntry> {
    return this.http.post<CalendarEntry>(this.baseUrl, entry);
  }

  updateCalendarEntry(entry: CalendarEntry): Observable<CalendarEntry> {
    return this.http.put<CalendarEntry>(`${this.baseUrl}/${entry.id}`, entry);
  }

  deleteCalendarEntry(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
